src/Shipping/ShippingZone.php

```diff
@@ -59,423 +72,112 @@ public function __construct( WC $wc, GoogleHelper $google_helper ) {
+	public function get_shipping_rates_for_country( string $country_code ): array {
 		$this->parse_shipping_zones();
 
-		return ! empty( $this->methods_countries[ $country_code ] ) ? array_values( $this->methods_countries[ $country_code ] ) : [];
+		$location_rates = $this->location_rates[ $country_code ];
+		if ( empty( $location_rates ) ) {
+			return [];
+		}
+
+		// Process the rates for each country subdivision separately.
+		$location_rates = array_map( [ $this->rates_processor, 'process' ], $location_rates );
+
+		// Convert the string array keys to integers.
+		$country_rates = array_values( $location_rates );
+
+		// Flatten and merge the country shipping rates.
+		$country_rates = array_merge( [], ...$country_rates );
+
+		return array_values( $country_rates );
 	}
 
 	/**
 	 * Returns the available shipping rates for a country.
 	 *
-	 * @param string $country_code
-	 *
-	 * @return array[] A multidimensional array of shipping rates for the given country. {
-	 *     Array of shipping method arguments.
+	 * If there are separate rates for the country's subdivisions (e.g. state,province, postcode etc.), they will be
+	 * grouped by their parent country.
 	 *
-	 *     @type string $country  The shipping method country.
-	 *     @type string $method   The shipping method ID.
-	 *     @type string $currency The currency which the shipping rate is in. Defaults to the store currency.
-	 *     @type float  $rate     The cost of the shipping method. Only if the method is flat-rate.
-	 *     @type array  $options  Array of options for the shipping method (varies based on the method type). {
-	 *         Array of options for the shipping method.
-	 *
-	 *         @type array[] $shipping_class_rates An array containing the shipping class names and their rates. Only if the method is flat-rate.
-	 *         @type float   $free_shipping_threshold The minimum order amount required to use the shipping method. Only if the method is free shipping.
+	 * @param string $country_code
 	 *
-	 *     }
-	 * }
+	 * @return LocationRate[]
 	 */
-	public function get_shipping_rates_for_country( string $country_code ): array {
-		$methods = $this->get_shipping_methods_for_country( $country_code );
-
-		// If the only shipping method is the free shipping method, return it as the only rate.
-		if ( 1 === count( $methods ) && self::METHOD_FREE === $methods[0]['id'] ) {
-			$rate = [
-				'country'  => $country_code,
-				'method'   => $methods[0]['id'],
-				'currency' => $methods[0]['currency'],
-				'rate'     => 0,
-				'options'  => [],
-			];
-			if ( isset( $methods[0]['options']['min_amount'] ) ) {
-				$rate['options']['free_shipping_threshold'] = $methods[0]['options']['min_amount'];
-			}
+	public function get_shipping_rates_grouped_by_country( string $country_code ): array {
+		$this->parse_shipping_zones();
 
-			return [ $rate ];
+		$location_rates = $this->location_rates[ $country_code ];
```

Comment:
When requesting the suggestions from the API I'm getting a notice on this line:
```
PHP Notice:  Undefined index: NL in wp-content/plugins/google-listings-and-ads/src/Shipping/ShippingZone.php on line 120
```

NL has a shipping zone and a rate, but it's skipped because it's `10 * [qty]`.

Reply:
This should be fixed by https://github.com/woocommerce/google-listings-and-ads/pull/1341/commits/21ad55f085c0d60ff1e0a357d12a3fdc833411fe